Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -615,13 +631,16 @@ class ArbitraryMipmapDetector
             const int diff_squared = diff * diff;
             pixel_diff += diff_squared;
           }
-          average_diff += pixel_diff;
+          current_diff_sum += pixel_diff;
         }
         ptr1 += shape.row_length;
         ptr2 += shape.row_length;
       }
+      // calculate the MSE over all pixels, divide by 2.56 to make it a percent
+      // (IE scale to 0..100 instead of 0..256)
```

Comment:
Not all formats uses 8 bit per pixel, eg the I4 format will get a much bigger error. Also the common rgb565 format will see a greater error. Do you think it is worth to add a table of texture formats and their precision?

Reply:
In my testing, the error on these texture formats was well within acceptable margins.

Reply:
That's an interesting thought, possibly something to look into later? I didn't see any issues with it in my testing. And if this code has an issue with those formats, the old code likely did too.